Cleanup of CDVDMAN/CDVDFSV

uyjulian opened the pull request:
## Pull Request checklist

Note: these are not necessarily requirements

- [x] I reformatted the code with clang-format
- [ ] I checked to make sure my submission worked
- [x] I am the author of submission or have permission from the original author
- [ ] Requires update of the PS2SDK
- [ ] Requires update of the gsKit
- [ ] Others (please specify below)

## Pull Request description

Cleanups to CDVDMAN/CDVDFSV, mainly to make it easier to work with.
Source files have been split up and definitions have been changed to the ps2sdk ones.

uyjulian committed:
Initial splitting of source for cdvdman and cdvdfsv

Use cd_file_t type instead of cdl_file_t, and move the extra flag entry into the RPC structure

Add ifdef guards on some include files

Move cdvdman default settings to ifdefs

uyjulian committed:
Change definitions to use the ps2sdk definitions where possible

uyjulian committed:
Change smsutils functions to use definitions to the original symbol

This will make it easier to change it out to use sysclib when it is ready

AKuHAK reviewed (commented):
Also 2 new warnings appear in the compilation log:
```sh
mipsel-ps2-irx-gcc -D_IOP -fno-builtin -Os -G0 -Wall -I../../network/common -I../common -I/usr/local/ps2dev/ps2sdk/iop/include -I/usr/local/ps2dev/ps2sdk/common/include -I. -Iinclude/ -DSMB_DRIVER -D__USE_DEV9 -msoft-float -mno-explicit-relocs -c device-smb.c -o obj.smb/device-smb.o
device-smb.c: In function 'DeviceFSInit':
device-smb.c:84:31: warning: taking address of packed member of 'struct cdvdman_settings_smb' may result in an unaligned pointer value [-Waddress-of-packed-member]
   84 |         smb_OpenAndX(tmp_str, &cdvdman_settings.FIDs[i++], 0);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
device-smb.c:93:35: warning: taking address of packed member of 'struct cdvdman_settings_smb' may result in an unaligned pointer value [-Waddress-of-packed-member]
   93 |             smb_OpenAndX(tmp_str, &cdvdman_settings.FIDs[i], 0);
      |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~
```

uyjulian committed:
Fix casting unaligned access warning

uyjulian committed:
Remove unneeded lines

uyjulian committed:
Change SCMD/NCMD enumerations to use the same ones in libcdvd in ps2sdk

uyjulian committed:
Change sector size '11' bitshifts to '2048' multiply/divide

This should make it more clear that it is an operation related to the sector size.
The compiler should optimize it to bitshifts anyway
